### DISCUSSION TITLE
Enhancement Request: Toggle Preference Functionality (Queries) and UI Elements (Units)

### DISCUSSION THREAD

**Discussion by lux4rd0** [grafana/grafana#38274] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

I want to have a dashboard that allows a user to choose how functionality and elements can change based on defined preferences.

**Why is this needed**:

I am specifically looking at a use case of metrics vs. imperial mathematical functions and UI elements. I have dozens of queries that return results in a metric format that I'm using math functions and Overrides Units to change to imperial values. With a dozen dashboards and hundreds of queries, I currently maintain multiple sets of dashboards, one metric and one imperial.

It would be extremely valuable to have a toggle (or drop-down) that sets a parameter or condition to determine "if this then that" for queries and unit selections.

`SELECT mean("air_temperature") * 9/5 + 32 FROM "weatherflow_obs" WHERE $timeFilter GROUP BY time($__interval)`

that has a Unit of "`Fahrenheit (°F)`"

but when toggled, would have a query of:

`SELECT mean("air_temperature") FROM "weatherflow_obs" WHERE $timeFilter GROUP BY time($__interval)`

with a Unit of "`Celsius (°C)`"

Dynamically "hiding" or "unhiding" may be an option for queries. However, for me, this would require more than the current limitation of 26 queries to be defined per panel (some tables have 14 queries already).

Another use case of this would be when using Overrides for values - I'm using `data:image/svg+xml` replacement to display icons in table views instead of text and are specific for a black background (white on transparent). A lookup for knowing the preference for dark/light UI themes would swap the image for one that works on a light background (black on transparent).

These two distinctions for toggles now require four times the number of dashboards to be managed versus one set with toggles.

An example of units to swap:

- temp: f/c
- wind: mph/kph/kts/mps/bft/ifm
- precip: in/mm/cm
- pressure: mb/inhg/mmhg/hpa
- distance: mi/km

![unit examples](https://user-images.githubusercontent.com/30187533/117042252-563e5500-acd1-11eb-8434-376c213ee36d.png)

_Edit: Note: while I feel this is a beneficial feature - when using dashboards as code, the ability to save these preferences under a user's individual preference (when logged in) would be amazing (rather than the saved dashboard default, which can't be overwritten.) Toggling every time a user accesses a dashboard to set their toggle preference would be more annoying from a usability standpoint._

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
